pypdf/_reader.py

```diff
@@ -2071,6 +2071,74 @@ def xfa(self) -> Optional[Dict[str, Any]]:
                         retval[tag] = es
         return retval
 
+    def add_form_topname(self, name: str) -> Optional[DictionaryObject]:
+        """
+        Add a top level form that groups all form fields below it.
+
+        Args:
+            name: text string of the "/T" Attribute of the created object
+
+        Returns:
+            The created object. `None`` means no object can be created.
+        """
+        catalog = cast(DictionaryObject, self.trailer[TK.ROOT])
+
+        if "/AcroForm" not in catalog or not isinstance(
+            catalog["/AcroForm"], DictionaryObject
+        ):
+            return None
```

Comment:
What do you think about raising a `MissingElementException("/AcroForm not in catalog")` instead of returning `None`?

Reply:
This cover the case where the form is not a form. I considered that the function should not raise an error

Reply:
What is a form that is not a form? Did you want to write "AcroForm"?

Does that mean that this grouping only makes sense for AcroForm? (I don't understand the AcroForm parts yet)